### PR TITLE
[leone] Updating ESyS

### PR DESCRIPTION
**Pull request by gppezzi**
- Updated `$PYTHONPATH` as requested by the user
- Adding Python as explicit dependency and using `pyver/pyshortver` variables

### CODE REVIEW

File: easybuild/easyconfigs/e/ESyS/ESyS-2.3.5-foss-2016b-Python-2.7.12.eb [hunk 1]
```diff
@@ -1,8 +1,10 @@
+# @author: jgp, gppezzi (CSCS)
```
Comment (jgphpc) [inline easybuild/easyconfigs/e/ESyS/ESyS-2.3.5-foss-2016b-Python-2.7.12.eb]: why not using `git blame` ?

File: easybuild/easyconfigs/e/ESyS/ESyS-2.3.5-foss-2016b-Python-2.7.12.eb [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
+
 easyblock = 'ConfigureMake'
 
 name = 'ESyS'
 version = '2.3.5'
-versionsuffix = '-Python-2.7.12'
+versionsuffix = '-Python-%(pyver)s'
```
Comment (jgphpc) [inline easybuild/easyconfigs/e/ESyS/ESyS-2.3.5-foss-2016b-Python-2.7.12.eb]: `pyver` is defined by the dependency with ('Python', '2.7.12') right ?

Comment (gppezzi) [inline easybuild/easyconfigs/e/ESyS/ESyS-2.3.5-foss-2016b-Python-2.7.12.eb]: Yes. But I don't know if that is a bug or feature..

Comment (gppezzi) [inline easybuild/easyconfigs/e/ESyS/ESyS-2.3.5-foss-2016b-Python-2.7.12.eb]: (in the sense that Python is already loaded by Boost, but `pyver` is not defined in this case)

**Comment by jgphpc** [comment]
jenkins failed because dom is down but leone passed ok:
https://jenkins.cscs.ch/job/ProductionEBtestingPR/191/console
=> merging.

File: easybuild/easyconfigs/e/ESyS/ESyS-2.3.5-foss-2016b-Python-2.7.12.eb (hunk 1 above)
Comment (gppezzi) [inline easybuild/easyconfigs/e/ESyS/ESyS-2.3.5-foss-2016b-Python-2.7.12.eb]: we can use `git blame` as a last resort, but it is nicer if people claim authorship. Besides, the latter option will also work if the file is moved to another git repo.

**Pull request merged**